### PR TITLE
chore(Documentation): Foundation Documentation

### PR DESCRIPTION
**Pull request by codecaaron**
- Adds knobs and editable versions to flex grid and layout grid.
- Add dedicated stories for base components: FlexGrid, Row, Col, Container (FlexBox).
- Refactor Color Stories to work better with Docs Mode.
- Remove some custom styling for things (but not all).

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eab3259a0ab90723f584a4c--codecademy-styleguide.netlify.app>https://5eab3259a0ab90723f584a4c--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eab3259a0ab90723f584a4c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.3.1-alpha.63d973.0<br/>@codecademy/gamut@7.4.3-alpha.63d973.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.3.1-alpha.b9c5eb.0<br/>@codecademy/gamut@7.4.3-alpha.b9c5eb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eab363ea0ab908064584bed--codecademy-styleguide.netlify.app>https://5eab363ea0ab908064584bed--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eab363ea0ab908064584bed>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.3.1-alpha.76443e.0<br/>@codecademy/gamut@7.4.3-alpha.76443e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eab42337b6cc3812071f087--codecademy-styleguide.netlify.app>https://5eab42337b6cc3812071f087--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eab42337b6cc3812071f087>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.3.1-alpha.9e9717.0<br/>@codecademy/gamut@7.4.3-alpha.9e9717.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eab51c3fd1c5c0f8c18858f--codecademy-styleguide.netlify.app>https://5eab51c3fd1c5c0f8c18858f--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eab51c3fd1c5c0f8c18858f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.3.1-alpha.667406.0<br/>@codecademy/gamut@7.4.3-alpha.667406.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eab9450f75dc4953574ffe2--codecademy-styleguide.netlify.app>https://5eab9450f75dc4953574ffe2--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eab9450f75dc4953574ffe2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.3.1-alpha.803700.0<br/>@codecademy/gamut@7.4.3-alpha.803700.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eac3576f75dc41dec7504e6--codecademy-styleguide.netlify.app>https://5eac3576f75dc41dec7504e6--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eac3576f75dc41dec7504e6>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.4.1-alpha.47bd31.0<br/>@codecademy/gamut@7.4.3-alpha.47bd31.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eac3aaba0ab904df2584e9c--codecademy-styleguide.netlify.app>https://5eac3aaba0ab904df2584e9c--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eac3aaba0ab904df2584e9c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eac3cab9311a76f3603e981--codecademy-styleguide.netlify.app>https://5eac3cab9311a76f3603e981--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eac3cab9311a76f3603e981>Deploy Logs</a></p>

File: packages/styleguide/stories/Foundations/Colors/Elements.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        justify="center"
+      >
+        <Container
+          column
+          className={styles.swatchDetail}
+          align="center"
+          justify="spaceAround"
+        >
+          <span className={styles.name}>{name}</span>
+          <CopyButton
```
Comment (kipprice) [inline packages/styleguide/stories/Foundations/Colors/Elements.tsx:55]: what are your thoughts on only enabling the copy button for non-deprecated colors?

Comment (codecaaron) [reply packages/styleguide/stories/Foundations/Colors/Elements.tsx:55]: 😮There's an idea.

Or just removing the deprecated colors 😎

Comment (JoshuaKGoldberg) [reply packages/styleguide/stories/Foundations/Colors/Elements.tsx:55]: _I'm working on it_ 😛

File: packages/styleguide/stories/Foundations/Colors/styles.module.scss [hunk 2]
```diff
@@ -35,19 +35,47 @@ h2.headingDark {
...
+  font-weight: 700;
+  font-size: 13px;
+  font-family: $font-family-headings;
+  opacity: 0;
+  transition: opacity 0.5s;
+}
+
+.swatchDetail {
+  height: 100%;
+  &:hover {
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/Colors/styles.module.scss:58]: This way we can <kbd>Tab</kbd> through them:

```suggestion
  &:focus,
  &:hover {
```

File: packages/styleguide/stories/Foundations/Colors/Elements.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            className={styles.swatchButton}
+            flat
+            theme="white"
+            caps
+            outline
+            copyValue={name}
+          >
+            Copy Name
+          </CopyButton>
+          <span className={styles.hexcode}>{hex}</span>
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/Colors/Elements.tsx:65]: I'm finding it hard to read lighter variants. Do you think they could have a drop shadow or background color or something like that?

Screenshot of a lighter variant with the white hex text

Also, getting the `color-xyz-####` SCSS name from the swatch view is a use case for me and I think others. Up for adding that back in?

File: packages/styleguide/stories/Foundations/Colors/Elements.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  return (
+    <>
+      {Object.keys(data).map(variableSuffix => {
+        const sassVariableName = getSassVariableName(
+          variablePrefix,
+          variableSuffix
+        );
+
+        const hexcode =
+          data[variableSuffix] === 'whitesmoke'
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/Colors/Elements.tsx:85]: this hurts me every time I see it...

File: packages/styleguide/stories/Foundations/Colors/Elements.tsx (hunk 3 above)
Comment (JoshuaKGoldberg) [reply packages/styleguide/stories/Foundations/Colors/Elements.tsx:65]: Oh wait they're there, just hidden!

Screenshot of searching for 'purple' and seeing it highlighted on purple swatches

File: packages/styleguide/stories/Foundations/Colors/Elements.tsx (hunk 4 above)
Comment (JoshuaKGoldberg) [reply packages/styleguide/stories/Foundations/Colors/Elements.tsx:85]: Actually, I think I missed this in the last cleanup, we don't have `whitesmoke` anywhere. Up for removing it? 😄

File: packages/styleguide/stories/Foundations/Colors/Colors.stories.tsx [hunk 5]
```diff
@@ -95,34 +47,53 @@ export const Colors = decoratedStory(() => (
       unsurprisingly referred to as "brand" colors. These are commonly used on
       buttons and as vibrant background colors.
     </StoryDescription>
-    <LayoutGrid className={styles.swatchesContainer} rowGap="md">
+    <LayoutGrid className={styles.swatchesContainer} rowGap="md" columnGap="md">
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/Colors/Colors.stories.tsx:50]: Now that these don't cause page stretches, they collapse in on each other in less wide viewports:

Screenshot of a thin viewport with overlapping column titles

Up for using your fancy responsive sizing properties?

File: packages/styleguide/stories/Foundations/Layouts/Flex/FlexGrid.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+  decoratedStory,
+  StoryTemplate,
+  StoryStatus,
+  StoryDescription,
+} from '../../../Templating';
+import { Box, Container } from '../Elements';
+
+export default decoratedStories('Foundations', 'Layouts', 'Flex', FlexGrid);
+
+export const fixedGridDefault = decoratedStory('Fixed Grid (Default)', () => (
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/Layouts/Flex/FlexGrid.stories.tsx:15]: I'm finding it hard as someone who's used these recently to parse through the differences between the `Foundations` > `Layouts` children. A few top-of-mind questions:

* Why is there both `Flex` > `FlexGrid` and `Grid` > `LayoutGrid`?
    * Suggestion: rename `Grid` to `LayoutGrid`, move `Basics` and `Editable` up, and rename `Basics` to `About`?
* What are the differences between them, and when/why would you use one over the other?
    * Suggestion: document when you would use one or the other in their parent `About` page, and include a little _"if you're looking for..."_ blurb in each of them pointing to the other?
* What is `Utilities` > `Container`?

File: packages/styleguide/stories/Foundations/Colors/Elements.tsx (hunk 3 above)
Comment (codecaaron) [reply packages/styleguide/stories/Foundations/Colors/Elements.tsx:65]: The idea was to make them searchable, and then you could click to copy.  Maybe I can mess around and let you choose the format.  `SCSS` | `JS` | `Hex`?

Little less transparent but those var names were extra long.  Theres some more work to be done to move this over to storybook palletes but we'll get there.

File: packages/styleguide/stories/Foundations/Layouts/Flex/FlexGrid.stories.tsx (hunk 6 above)
Comment (codecaaron) [reply packages/styleguide/stories/Foundations/Layouts/Flex/FlexGrid.stories.tsx:15]: I'm gonna hold off on this part till we get the MDX stuff rolling.  This was mostly hacks and MDX is just way better for this and will cut down on the amount of folders we have.

File: packages/styleguide/stories/Foundations/Colors/Colors.stories.tsx (hunk 5 above)
Comment (codecaaron) [reply packages/styleguide/stories/Foundations/Colors/Colors.stories.tsx:50]: Gotchu

File: packages/styleguide/stories/Foundations/Colors/Elements.tsx (hunk 3 above)
Comment (codecaaron) [reply packages/styleguide/stories/Foundations/Colors/Elements.tsx:65]: I added a text-shadow.  It's not perfect, but at least it's clear that it's present.  I think we should probably use the storybook builtins for this anyways.

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eb066f915a72597dee88227--codecademy-styleguide.netlify.app>https://5eb066f915a72597dee88227--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eb066f915a72597dee88227>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eb069b298c8ba018620c7d0--codecademy-styleguide.netlify.app>https://5eb069b298c8ba018620c7d0--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eb069b298c8ba018620c7d0>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5eb076fcc120477c51c9c7ea--codecademy-styleguide.netlify.app>https://5eb076fcc120477c51c9c7ea--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5eb076fcc120477c51c9c7ea>Deploy Logs</a></p>

**Pull request merged**